syntax = "proto3";

package protodb.collection;

import "google/protobuf/descriptor.proto";

message CreateCollectionRequest {
    string database = 1;
    string name = 2;
    google.protobuf.DescriptorProto schema = 3;
}

message CreateCollectionResponse {
    bool success = 1;
    enum FailureCode {
        NO_FAILURE = 0;
        INVALID_DATABASE = 1;
        COLLECTION_EXISTS = 2;
        SCHEMA_ERROR = 3;
    }
    FailureCode failure_code = 2;

    message SchemaError {
        enum SchemaErrorCode {
            NO_SCHEMA_ERROR = 0;
            MISSING_ID_FIELD = 1;
            INVALID_ID_TYPE = 2;
            INVALID_FIELD_TYPE = 3;
        }
        SchemaErrorCode code = 1;
        string message = 2;
    }
    SchemaError schema_error = 3;
}